## So you want to do research on Twitter

This tutorial covers the basics of using the _twitter_ library in Python to perform searches using the REST API.  

__NB:__ There are a lot of different ways of accessing and processing Twitter data. I'm going to go through my workflow, which is a kludging together of methods assembled over the last four years.  There are probably better approaches to at least some parts of what I'm going through here, and if you know of one or find one, let me know!

## What's Twitter like?

Not to sound like a tech evangelist, but one neat thing about Twitter is that everyone uses and experiences it differently.  (Let's have an aside about everyone's familiarity with Twitter and what they use it for.)

Now, here's the interesting thing to me about using Twiter for research: Twitter is much more mundane than you think. It's a little paradoxical; most tweets on Twitter are nothing like most tweets that we see, because the tweets that pull big numbers (views, favs, RTs) are from bigger accounts and/or disseminated through the network broadly. But most tweets, like most of us, die in obscurity.

Here's an example of this.  Let's search for something really pragmatically empty, so we're going to get a scattering of topics. Specifically, let's look at tweets containing the word *the*:

https://twitter.com/search?l=&q=the&src=typd

The usual, common suspects show up in the default web-based search. Trump, 5 Seconds of Summer, news organizations, etc.  But that's because the Twitter algorithm defaults to show you a mix of recent and "top" tweets.  If you click on the little "Latest" tab, it gives you all the recent tweets for *the*, and it looks a lot different. 

https://twitter.com/search?f=tweets&vertical=default&q=the&src=typd

Here are the normies, the folks like you or me, or at least me, who chat with a few friends or try in vain to get noticed. Now, depending on your goals in your research, you might want the popular accounts and the viral tweets.  But if you're interested in how people talk, or how they interact, it's these normies who are your best friends.

PS: if you want to test out your searches before APIing them, here's the advanced search link: https://twitter.com/search-advanced


## Initialization basics

We'll start by setting up a Twitter search object, and doing a basic search.

In [9]:
import sys, os
from pprint import pprint                           #Important for reading through JSONs
from time import localtime,strftime,sleep,time      #Important for dealing with Twitter rate limits
import twitter

### Authorizing your account

The first thing you need is to perform an OAuth dance to tell Twitter who you are, and that you're cool. This is a slightly complicated step, involving identification both of a user conducting the searches and a client through which they're conducting them.

Hopefully you have already set up both your Twitter account and an app for accessing this data. If not, do it now; it's a pretty trivial process once you have a Twitter account, and can be set up in a few minutes here: https://apps.twitter.com/app/new

[Optional aside with link to set up app.]

Now that you have an app set up, we're going to input its access keys. Go to https://apps.twitter.com, click on your app, and then click on the "Keys and Access Tokens" tab.  Copy them in when prompted.

In [2]:
cons_oauth_file = 'c.xxx'
if os.path.exists(cons_oauth_file):
    constoken, conssecret = twitter.read_token_file(cons_oauth_file)
else:
    constoken = raw_input("What is your app's 'Consumer Key'?").strip()
    conssecret = raw_input("What is your app's 'Consumer Secret'?").strip()
    wf = open(cons_oauth_file,'w'); wf.write(constoken+'\n'+conssecret); wf.close()

Now we'll authorize this app through your Twitter account. This code block will open a new tab, prompt to activate the app, and provide you with a PIN to enter back here. This OAuth registration is handled by the *twitter.oauth_dance* function.

In [3]:
app_oauth_file = 'a.xxx'
if not os.path.exists(app_oauth_file):									#if user not authorized already
	twitter.oauth_dance("your app",constoken,conssecret,app_oauth_file)		#perform OAuth Dance
apptoken, appsecret = twitter.read_token_file(app_oauth_file)					#import user credentials

### Initializing a search object

Here's the core of the code, really -- we're setting up an object that uses your credentials to access the Twitter REST API and return structured Twitter data.  Thanks to the _twitter_ library, it's a single command, and then we can work with the resulting object, _tsearch_.

In [4]:
tsearch = twitter.Twitter(auth=twitter.OAuth(apptoken,appsecret,constoken,conssecret))	#create search command

## Basics of the search object & API

Now that _tsearch_ is initialized, let's get to searching! Except, actually, let's hold up a second and talk about how to use the API.

Twitter provides fairly extensive (but often confusing) documentation for its REST API, available here: https://dev.twitter.com/rest/reference  Basically, the REST API handles searches of recent tweets (the last week, generally), as well as functions related to the social network structure, trending topics, and user information.

### An aside: what won't we be covering?

There is another Twitter API, the streaming API, which maintains a connection to Twitter and collects tweets as they occur.  I don't have enough experience with the streaming API to teach it, and many of its potential applications can be handled through the REST API.  I also won't be dealing with the POST features of the REST API, because we're focused on collecting existing data rather than generating our own.

### Rate limits

Perhaps the most important thing to know about for the REST API is that it is rate-limited. For each app/account combo, you can only perform so many searches per time period. I believe Twitter resets its counter every 15 minutes for all API calls; it definitely does for all the common ones.  If you try to search once you've hit the rate limit, the search will return an error. You will want to make sure your code doesn't persist much after it hits a rate limit, or else Twitter may get mad at your account (this is why I suggest setting up an account specifically for these searches, just in case).

There is also a return limit, both as a limit to the number of results any single search will give you, and a limit on how far back Twitter will search for you.  In general, you can't ask for more than 100 hits per individual search, though you can search reatedly to get more and more 100-hit blocks.  But after a while, Twitter says enough of that and stops giving you anything new.  We'll get to illustrations in a little bit, but for now, here are some commonly encountered limits, and the full list is available [here](https://dev.twitter.com/rest/public/rate-limits).

| Function call | limit per 15 minutes | max returns per search | max history | 
|---|---|---|
| GET search/tweets | 180 | 100 | 3000 / 7 days |
| GET statuses/user_timeline | 900 | ? | 3200 / infinite past |
| GET users/lookup | 900 | 100 | NA |

### Checking rate limits

Since we don't want to anger Twitter, let's talk about checking rate limits.  Your _tsearch_ object can check rate limits, and it's a nice way of getting used to the nested list format of the search results.

By the way, checking rate limits **is itself rate limited** (180 / 15 minutes)! 

In [6]:
r = tsearch.application.rate_limit_status()  #Obtain rate limit status for all searches
pprint(r)

{u'rate_limit_context': {u'access_token': u'875840141905936384-mHcprEkKJs6WUpQ0ImHTJTctLWzRrZP'},
 u'resources': {u'account': {u'/account/login_verification_enrollment': {u'limit': 15,
                                                                         u'remaining': 15,
                                                                         u'reset': 1498163071},
                             u'/account/settings': {u'limit': 15,
                                                    u'remaining': 15,
                                                    u'reset': 1498163071},
                             u'/account/update_profile': {u'limit': 15,
                                                          u'remaining': 15,
                                                          u'reset': 1498163071},
                             u'/account/verify_credentials': {u'limit': 75,
                                                              u'remaining': 75,
                                

*application.rate_limit_status()* returns a set of nested dictionaries; one for each type of API request, sorted by request type. For each request, it lists the rate limit (*'limit'*), the number of API calls you have left before you hit that limit (*'remaining'*), and the Unix time at which your rate limit resets (*'reset'*).

Each API call type is tracked completely separately, and the rate-limit window is **15 minutes from your first call *specific to that request*.**  For instance, let's try doing a standard term-based search (GET search/tweets) and then check the rate limits again.

In [7]:
res = tsearch.search.tweets(q='test',              #A test query
                            count=10,              #just want 10 hits back
                            result_type="recent")  #include all recent tweets, not only popular ones

r = tsearch.application.rate_limit_status()  #Obtain rate limit status for all searches
pprint(r['resources']['application'])        #rate limit query's rate limit
pprint(r['resources']['search'])             #search query's rate limit

{u'/application/rate_limit_status': {u'limit': 180,
                                     u'remaining': 178,
                                     u'reset': 1498163071}}
{u'/search/tweets': {u'limit': 180, u'remaining': 179, u'reset': 1498163597}}


We've now done two rate limit queries, so the remaining available in this window is 178. But we've only done one term-based query, so there are still 179 left on that.  But importantly, the two have different reset times, because the rate-limit reset was triggered by the first search from the previous code block, while the term search was triggered more recently.

By the way, the reset time is always given in Unix time, UTC (i.e., number of milliseconds since 1/1/1970 UTC). Converting to local human-readable time isn't too bad, using *localtime* and *strftime* from the Python *time* library:

In [18]:
print('Rate limit reset:')
print(localtime(r['resources']['application']['/application/rate_limit_status']['reset']))
print(strftime('%I:%M:%S',localtime(r['resources']['application']['/application/rate_limit_status']['reset'])))
print('\n')
print('Term search reset:')
print(strftime('%I:%M:%S',localtime(r['resources']['search']['/search/tweets']['reset'])))

Rate limit reset:
time.struct_time(tm_year=2017, tm_mon=6, tm_mday=22, tm_hour=13, tm_min=24, tm_sec=31, tm_wday=3, tm_yday=173, tm_isdst=1)
01:24:31


Term search reset:
01:33:17


### Dealing with the rate limit

In this tutorial, we won't push up against the rate limit much, but for many tasks, you're going to have to deal intelligently with it. Let me share a few quick tricks, and then we'll move on to real searches.

**Trick 1: *time.sleep()* **

The most basic solution is to take advantage of the *sleep* function, which tells Python to hold on for some specified length of time. When your 'remianing' value drops to (or near) zero, record the 'reset' time, and subtract the current time from it (*time.time()*).  I generally add a little buffer time, too, in case my system clock is off from Twitter's.

One note: if you haven't triggered a rate limit window by doing a search, the reset value will be 15 minutes from now, so build in a check to make sure that you're not sitting around infinitely.  Any suggestions for how to best handle that?

In [22]:
print 'Reset time:  ',r['resources']['application']['/application/rate_limit_status']['reset']
print 'Current time:',time()
print 'Wait time:   ',r['resources']['application']['/application/rate_limit_status']['reset'] - time() + 30

Reset time:   1498163071
Current time: 1498165018.64
Wait time:    -1917.64221597


**Trick 2: interleave requests**

Because the rate limits are triggered separately for each search type, if you're doing multiple search types (say, term and user searches), you can switch back and forth, working with one until the limit's hit, then going to the other.

In [27]:
temp = tsearch.followers.list()
r = tsearch.application.rate_limit_status()
pprint(r['resources']['followers'])

{u'/followers/ids': {u'limit': 15, u'remaining': 15, u'reset': 1498165552},
 u'/followers/list': {u'limit': 15, u'remaining': 12, u'reset': 1498165478}}


**Trick 3: process in the moment**

If you're doing any substantial tweet processing within Python, you can take advantage of the rate limit to do time-consuming processing while you wait. Since the rate limit window is triggered by the first search, you want to make sure to put the first search as early in your code as possible, but subsequent searches can come either before or after processing wihtout consequnces.

**Pitfall 1: track limits internally, confirm occasionally**

One small thing: since the rate limit query is itself rate-limited, you want to ration your calls to it, especially if you're performing more than 180 searches by interleaving request types. I keep an internal count of the number of times I called each search, and then every 3 or so searches, check back with *application.rate_limit_status()* to make sure twitter and I are in agreement.

### What happens if we pass the rate limit?

All right, let's give in to our darker motivations and wantonly exceed the rate limit to find out the consequences.



In [33]:
r = tsearch.application.rate_limit_status()
for i in range(0,r['resources']['followers']['/followers/list']['remaining']):
    temp = tsearch.followers.list()

In [39]:
try:
    temp = tsearch.followers.list()
except:
    print "error"

error
{u'next_cursor': 0,
 u'next_cursor_str': u'0',
 u'previous_cursor': 0,
 u'previous_cursor_str': u'0',
 u'users': []}


Ah, it just kinda hangs there! So maybe that sets up:

**Trick 5: always *try* **

Unfortunately, the *twitter* library doesn't handle errors very well, in part, I think, because Twitter changes how the API handles errors from time to time.  So, if you're going to be doing a large number of searches and/or automating your system, be sure to build some safety valves in the form of *try-catch-except* statements around all the non-manual searches.

Note that sometimes the system itself craps out, so not all errors are due to rate limit violations. For important searches that I'm not observing, after a failed search, I have it wait a few seconds and try again (and maybe even another wait-try cycle), before doing a 15-minute wait cycle to make sure all the rate limits have refreshed.  You can probably work out a better system than that.

## Actually searching

Okay, enough of all the boring limitations. Let's take this puppy out on the road in Section 2!

In [2]:
#Global variables
loclist = ["30.8,-98.6","39.8,-95.6","32.8,-117.6","37.8,-122.6"]
radius = "2500km"
multiloc = True
tweetspersearch = 75
maxpages = 20
maxid = float("+inf")
importcsv = ''
overwrite = 'w'
outfile = ''
keeptweets = True
startat = 0
header = True
throttle = True
checklimits = 3 		#To avoid overquerying the rate_limit function, only check every N iterations
importmultiloc = False
newmultiloc = False
onlyincltweets = True
wff = False
trackfails = False
MAXERRORS = 3
scheduled = False
raw = True
tweetcount = 0

### Setting up your account to use SeeTweet

First things first, you need a Twitter account through which to access the REST API. Just in case you run afoul of any Twitter regulations---and it's not inconceivable as you start out---I recommend using an account you could survive being Twitter jailed.

In [3]:
tsearch = authorize()

### Choosing a search term

Okay, we're going to look for some term here. You can look for individual words or a phrase.

In [4]:
term1 = '"hot outside"'
term1 = re.sub(' ','+',term1)
term = re.sub('\"','%22',term1)

print "Searching for", term

Searching for %22hot+outside%22


### Main code

This will be revised to simply & improve transparency. Ideally will be reduced to simple API calls with reduced opaque processing.  Some additional processing I'm currently doing will also be removed.

Good feature to add here is URL for each discovered tweet in case you want to look at what's going on as it comes out.

In [5]:
#Basic startup processing
if (importcsv and multiloc):
	importmultiloc = True
elif multiloc:
	newmultiloc = True


if importcsv:
	rf = open(importcsv,'r')
	firstline=True
	tidnum = 0
	tids = []
	centers = []
	incls = []
	for line in rf:
		if line[0] == '#':
			continue
		splitline = line.strip().split(',')
		if firstline:
			tidnum = splitline.index('tid')
			if importmultiloc:
				centernum = splitline.index('center')
				inclnum = splitline.index('incl')
			firstline=False
		else:
			if onlyincltweets:							#If we're excluding excluded tweets from baseline calc, skip to next line if incl=0
				if int(splitline[inclnum])==0:
					continue
			tids.append(long(splitline[tidnum])-1)
			if importmultiloc:
				centers.append(int(splitline[centernum]))
				incls.append(int(splitline[inclnum]))
	tids = tids[startat:]
	if importmultiloc:
		centers = centers[startat:]
		incls = incls[startat:]
	maxpages = len(tids)
	tids.append(0)
	firsthit = tids[0]
	rf.close()
else:
	firsthit = float("+inf")

outcomes = {}
locnum = -1
if newmultiloc:
	tweetlist = [0]*len(loclist)
	searchesleft = [0]*len(loclist)
	mintids = [0]*len(loclist)
	maxtids = [0]*len(loclist)
elif importmultiloc:
	tweetlist = [0]*maxpages		#tweetlist[tweetnum][locnum][resnum] = [outline,loc,tid] for the resnum-th baseline tweet in loc locnum on testtweet tweetnum
	mintids = [0]*maxpages
	maxtids = [0]*maxpages
	

In [9]:
#Actual search-performing code
outfile = ''

if importcsv:
	if not outfile:
		outfile = 'base.'+term1.strip('\"')+'.'+importcsv
else:
	#if scheduled:
	#	scheddir = 'scheduled'
	#	if not os.path.exists(scheddir):
	#		os.mkdir(scheddir)
	#	outfile = scheddir+'/'+term1.strip('\"')+'.'+strftime('%Y%m%d.%H%M')+'.csv'
	if not outfile:
		outfile = term1.strip('\"')+'.csv'
wf = open(outfile,overwrite)
latlong = loclist[0]
#TODO: block out the header stuff
if header:
	wf.write('#Compiled by SeeTweet '+versionnum+'.\n')
	wf.write('#Search performed at '+strftime('%Y-%m-%d %H:%M')+'\n')
	if not multiloc:
		wf.write('#Search location: '+latlong+','+radius+'\n')
	else:
		wf.write('#Search location: U.S. 4-location coverage points (Texas, KC, SD, SF)\n')
	wf.write('#Search term: '+term1+'\n')
if (overwrite=='w' and not multiloc):
	wf.write('day,year,month,date,hour,minute,second,source,city,state,lat,long,uid,tid\n')
elif (overwrite=='w' and newmultiloc):
	wf.write('day,year,month,date,hour,minute,second,source,city,state,lat,long,uid,tid,center,incl\n')
elif (overwrite=='w' and importmultiloc):
	wf.write('day,year,month,date,hour,minute,second,source,city,state,lat,long,uid,tid,origtid,origincl,center,incl\n')
if keeptweets:
	tweetdir = 'tweetarchive'
	if not os.path.exists(tweetdir):
		os.mkdir(tweetdir)
	#if not os.path.exists(tweetdir+'/'+scheddir):
	#	os.mkdir(tweetdir+'/'+scheddir)
	outtweetfile = tweetdir+'/'+os.path.splitext(outfile)[0]+'.tweets'
	wft = open(outtweetfile,overwrite)
	if overwrite=='w':
		if raw:
			wft.write('day\tyear\tmonth\tdate\thour\tminute\tsecond\tloc\tuid\ttid\ttweet\ttlength\trt\n')
		else:
			wft.write('loc,uid,tid,tweet\n')
if trackfails:
	faildir = 'failures'
	if not os.path.exists(faildir):
		os.mkdir(faildir)
	outfailfile = faildir+'/'+os.path.splitext(outfile)[0]+'.fails'
	wff = open(outfailfile,'w')
	
	
	#Note: block out the rate check
for latlong in loclist:
	locnum = locnum + 1
	geocodestr = latlong+","+radius
	print "\nSearch centered at:", latlong, "(locnum "+str(locnum)+")"
	if newmultiloc:
		currloctweets = []
		tidbycurrloc = []
	
	for pagenum in range(0,maxpages):
		print ''
		if importmultiloc:
			if (locnum==0):
				tweetlist[pagenum] = []
				maxtids[pagenum] = []
				mintids[pagenum] = []
			#print maxtids
			currloctweets = []
			tidbycurrloc = []
		#Examining the rate limit
		if (pagenum % checklimits == 0):
			r = getlimits(tsearch)
			if r['remaining'] <= checklimits:
				print "\n\n**Paused because of rate limit.**"
				print "Current time:",strftime('%I:%M:%S')
				print "Reset time:  ",strftime('%I:%M:%S',localtime(r['reset']))
				if not multiloc:
					print "Resume with flag -s="+str(startat+pagenum)
				else:
					print "Stopped on location "+str(locnum)+", tweet "+str(startat+pagenum)+"/"+str(maxpages)
				print "--"
				print tweetcount, 'tweets found. Centered at', geocodestr
				print outcomes
				if throttle:
					waittime = r['reset']-time()+30
					print "Waiting", round(waittime), "seconds before resuming."
					sleep(waittime)
				else:
					sys.exit()
			elif r['remaining'] < 11:
				print "\n\n**WARNING:", r['remaining'], "queries remaining.**"
				print "Current time:",strftime('%I:%M:%S')
				print "Reset time:  ",strftime('%I:%M:%S',localtime(r['reset']))
				print ""
				print "\n\n"
				sleep(10)
		
		#Adding a catch for various Twitter errors
		errors = 0
		while (errors < MAXERRORS):
			try:
				if (firsthit == float("+inf")):
					res = tsearch.search.tweets(q=term+'+-rt',geocode=geocodestr,count=str(tweetspersearch),result_type="recent")
					#pprint(res['statuses'][99])
				else:
					#print 'Test:', term+'+-rt',geocodestr,str(tweetspersearch),str(firsthit)
					res = tsearch.search.tweets(q=term+'+-rt',geocode=geocodestr,count=str(tweetspersearch),result_type="recent",max_id=str(firsthit))
					#pprint(res['statuses'][99])
				break
			except TwitterHTTPError as e:
				errors = errors + 1
				print "Twitter Error encountered. Retrying",MAXERRORS-errors,"more times."                                                       
				print "\n"+e.response_data
				sleep(5)
		if (errors==MAXERRORS):
			print "Repeated errors encountered, possibly due to rate limit."
			print "Will wait 15 minutes and try once more before quitting."
			sleep(900)
			try:
				if (firsthit == float("+inf")):
					res = tsearch.search.tweets(q=term+'+-rt',geocode=geocodestr,count=str(tweetspersearch),result_type="recent")
				else:
					res = tsearch.search.tweets(q=term+'+-rt',geocode=geocodestr,count=str(tweetspersearch),result_type="recent",max_id=str(firsthit))
			except:			
				raise Exception("Gave up because of repeated errors, sorry.")
		res = res['statuses']
			
		print len(res), 'hits on page', startat+pagenum+1, '(max_id='+str(firsthit)+')'
		print r['remaining']-1, 'queries remaining.'
		if (len(res)==0):
			print 'Out of tweets at this location.'
			#print 'Test:', term+'+-rt',geocodestr,str(tweetspersearch),str(firsthit)
			#sleep(1)
			#res = tsearch.search.tweets(q=term+'+-rt',geocode=geocodestr,count=str(tweetspersearch),result_type="recent",max_id=str(firsthit))
			#pprint(res['statuses'][99])
			break
		for i in range(0,len(res)):
			tweetcount = tweetcount + 1
			[outline,outcome,tid,tline] = extractinfo(res[i],wff,raw)
			if outline:
				if not multiloc:
					wf.write(outline)
				elif newmultiloc:
					currloctweets.append([outline[:-1]+','+str(locnum)+'\n',locnum,long(tid)])
					tidbycurrloc.append(float(tid))
					#wf.write(outline[:-1]+','+str(locnum)+'\n')
				elif importmultiloc:
					currloctweets.append([outline[:-1]+','+str(tids[pagenum]+1)+','+str(incls[pagenum])+','+str(locnum)+'\n',locnum,long(tid)])
					tidbycurrloc.append(float(tid))
			if keeptweets:
				#wft.write(tline.encode('ascii','ignore'))
				if not raw:
					wft.write(tline.encode('ascii','ignore'))
				if raw:
					wft.write(tline.encode('ascii','replace'))		#maybe create a separate .utweets file with the Unicode versions of tweets
			outcomes[outcome] = outcomes.get(outcome,0)+1
			if importcsv:
				firsthit = tids[pagenum+1]
			else:
				if firsthit > long(tid):			#if current tweet came before previous oldest, update oldest
					firsthit = long(tid)-1
		if importmultiloc:
			if len(tidbycurrloc) > 0:
				maxtids[pagenum].append(max(tidbycurrloc))
				mintids[pagenum].append(min(tidbycurrloc))
			else:
				maxtids[pagenum].append(0)
				mintids[pagenum].append(0)				
			tweetlist[pagenum].append(currloctweets)
	#endfor searches within a location
	if newmultiloc:
		if len(tidbycurrloc) > 0:
			maxtids[locnum] = max(tidbycurrloc)
			mintids[locnum] = min(tidbycurrloc)
		else:
			maxtids[locnum] = 0
			mintids[locnum] = 0
		searchesleft[locnum] = maxpages-pagenum-1			#calculating how many pages were left in the most maxed-out search
		tweetlist[locnum] = currloctweets
	overwrite = 'a+'
	header = False
	if importcsv:
		firsthit = tids[0]
	else:
		firsthit = float("+inf")

#Endfor multiple locations
if newmultiloc:
	balanceandprint(tweetlist,mintids,maxtids,searchesleft,wf)
elif importmultiloc:
	for pagenum in range(0,maxpages):
		balanceandprint(tweetlist[pagenum],mintids[pagenum],maxtids[pagenum],[0]*len(loclist),wf)
wf.close()
if keeptweets:
	wft.close()
if trackfails:
	wff.close()
	


Search centered at: 30.8,-98.6 (locnum 0)

75 hits on page 1 (max_id=inf)
179 queries remaining.
CS: new york,ny 19 Breezy Outfits For When the Weather Is Just Too Hot to Handle https://t.co/op0MUB3MIB https://t.co/rXjpcvmUzL
CS: las vegas,nv It's hot outside, let me go swimming in your eyes
CS: shawnee,ok When it's hot outside and raining at the same time ..😩 https://t.co/9dUbLTT66v
F: toronto  ontario 736212357584654336
CS: las vegas,nv Watch Hard 4K Porn Videos For Free ➔ https://t.co/9lT4vOIYhU #porn #vip #movies #video #ass #cumtribute https://t.co/Nw5qPopxM7
F: minnesota 860550394006646784
CS: las vegas,nv Watch hard porn videos for free ➔ https://t.co/9lT4vOIYhU #hotgirl #hot #pussy #sext #videos #movies https://t.co/gcyWcApv0x
CS: phoenix,az Hot outside? Grab a refreshing... sewage popsicle? (Yuck; it's not really edible!) https://t.co/uShSeoBhAi https://t.co/3tIPYiKtFP
CS: phoenix,az Hot outside? Grab a refreshing... sewage popsicle? (Yuck; it's not really edible!) https://t.

CS: mountain home,ar It may be getting hot outside, but the HOTTEST deals are at Ultimate Chrysler Dodge Jeep RAM!!! https://t.co/u6FyKYdLZX
F: new york 16561306
CS: las vegas,nv adult vids in HD get your fREE access ➔ https://t.co/9lT4vOIYhU #nsfw #video #pussy #ass #movies #sexy #swallowcum https://t.co/37vs6zxbK9
F: colorado  usa 1066526090
F:  1145906497
CS: winter park,fl It's hot outside today. I'm slightly bored wanting a cold shower
CS: fontana,ca I feel like it's so hot outside & now I'm debating if I wanna dress like a thot today or put on actual clothes
F: canada 2601553358
F: utah  usa 447291916
F: jamaica 4214813592
F: midwest 362034298
CS: orlando,fl It's too damn hot outside
F: texas  usa 713197429
CS: scottsdale,az It's hot outside 👉 shade your head in style! https://t.co/Ycvp7gkE5x
CS: amarillo,tx It's way hot outside , I'm melting .
CS: austin,tx @nahitscooll It's hot outside!
F: overbrook  philadelphia 825706433278799872
LL: 27.220232,-80.795969 Ok it’s hot outside, 

CS: kansas city,mo It's hot outside an it looks like she's natural so it's possible that she could have sweated out her leave out 😑😑😑😑
CS: killeen,tx It may be getting hot outside, but don’t expect Killeen’s Pershing Park Pool to open up anytime soon. https://t.co/YUE3MoI7CP
F:  587198022
F: goderich ontario canada 172002568
KC: houston I'm deciding if I should go to pride this year or not it's gonna be so damn hot outside and I got no friends but idc
F: chicagoland usa 294907233
F: canada 3061926986
CS: mission,tx George always likes to go bike riding when it's really hot outside! I look darker😭
CS: chicago,il This that awkward weather where it's hot outside and freezing in every building
CS: kenedy,tx Dead ass sitting in the cooler at work bc it's so fkn hot outside 😭
CS: vinton,la it's so freaking hot outside
CS: grand rapids,mi 21 Cats That Are 100% You When It's Hot Outside https://t.co/5fRzJhl9kX
F: ottawa  canada 19487948
F: texas  usa 1678568611
CS: new orleans,la Why it gotta 

F: toronto  ontario 848340199826493441
CS: houston,tx So damn hot outside 😅
CS: san antonio,tx fuckin hot outside
CS: lexington,ky It's hot outside, but the sky is blue. I'd call that a double standard!
CS: minneapolis,mn @Dish_AWD BUT ITS HOT OUTSIDE!!
F: staten island 529078896
F: north carolina 4868361670
CS: wheaton,md It may be hot outside, but you can stay cool at the @WheatonIce  arena! https://t.co/EXBK6v88J6
CS: dallas,tx ⚠️it's getting quite hot outside so please do not leave anything that breathes in a hot car, even if it's just for a few minutes!!⚠️
F: avenue de la tristesse 295461578
CS: new orleans,la It's #sunny and #hot outside! It's the perfect day for a #snoball ! Find your favorite snoball flavor of... https://t.co/6GOLBY4LPG
F: toronto  ontario 79853685
F: 🗣dat985! 3048481119
F: bloomington  minnesota  usa 21969606
F: hill valley 353339455
KC: detroit @iamthedriving What is there to get? It's a park with slides and play sculptures that also has water, which is appe…

75 hits on page 8 (max_id=872578008699342851)
173 queries remaining.
KC: el paso It's so hot outside, let me float naked in your pool while drinking a 40oz.
CS: houston,tx @Jzacharyyrahcaz You right, us Mexicans were BORN to work lol.. but still to hot outside for me 🤷🏻‍♂️
CS: tampa,fl How to Manage Your Diabetes When It's Hot Outside https://t.co/NQkqNYqW6v https://t.co/UBOdstOQt3
F: toronto 162395712
CS: goldsby,ok It's hot outside, but we've got cold drinks and good music!! Open mic starts at 7 with Dwight Hamlin!! https://t.co/pDmhSBHogP
CS: tampa,fl How to Manage Your Diabetes When It's Hot Outside https://t.co/NQkqNYqW6v https://t.co/bdKy4ZUVnj
CS: spokane valley,wa It may be getting hot outside, but you can still have cool styles for less! Shop our 50% off Clearance sale NOW! https://t.co/SLeNXZ9J03
F:  57213222
CS: las vegas,nv Just a lil stinkin hot outside but McDonald's Fries sounded soo good! Fat, Full & happy🐷🐷🐷
F: kingston  jamaica 53140952
F: texas 2728610300
CS: tampa,f

CS: east peoria,il @JimGaffigan You dropped it. Dropped it like it's hot...outside. Inside, it's nice and cool. That's why the bugs try to get in.
CS: dallas,tx Getting hot outside 3 more hours to go
CS: potomac,md It's getting hot outside. Girls start acting up in this weather https://t.co/zA1BOIhSQk
CS: charlotte,nc @Pruny_thebeagle He's been doing good! Just trying to stay cool while its been hot outside.
CS: austin,tx It's so freaking hot outside!!
F: moon 105005407
CS: pataskala,oh Sheesh I need a gf . It's too hot outside to be single right now
CS: statesville,nc So hot outside.  Can't wait to get home Dx
KC: atlanta georgia usa POLL: #Summer heat got you beat? For #WorkoutWednesday, tell us how you stay fit when it’s hot outside!
F: california  usa 3238031588
CS: new york,ny It's the start of "it's balls hot outside but -10 degrees in your office" season
CS: miramar,fl When it's hot outside, head inside at #AtlanticoAtMiramar! We've got the clubhouse you want to relax in! #LiveB

CS: edinburg,tx It's so hot outside!😩😭
CS: refugio,tx It freaking hot outside!!!!
F: haltom city 587844473
CS: round rock,tx It's too hot outside 🔥
CS: dallas,tx listen... it's too damn hot outside
F: maryland 22300781
F: ohio 597987914
CS: phoenix,az Did anyone else know it's hot outside or is that just me?  I might have a magic temp gauge that is giving me... https://t.co/7RpkgiDFFB
CS: wylie,tx Really don't wanna walk in this store.. too damn hot outside 😭
F: tx 56600827
F: california  usa 828086201697628160
CS: hemet,ca It's already getting hot outside!!!
CS: houston,tx It be too hot outside for all that heavy makeup..
CS: houston,tx @StDxLe3Loo Pretty hot outside today haha
CS: longview,tx Man it's hot outside 🙋🏾‍♂️😒
CS: san antonio,tx @Stephanie_3HK Cause it's hot outside
F: avl 510823412
F: canada 20442664
CS: martinsville,va it's scorching hot outside
CS: memphis,tn It's so fucking hot outside 😩
KC: kansas city #HowIActuallySpendMySummer bitching because it's too freakin' hot o

KC: san antonio tx united states @chadzmom @MustStopMS It is amazing how alive I feel in the pool, even when it is Texas hot outside #ChatMS
CS: bakersfield,ca It's sooo fucking hot outside & I have to wait for my mom until she gets out of work & I don't have Fucking gas & I have leather seats😤
F: kansas  usa 2930792892
F: california 17637892
CS: corpus christi,tx It's to fucking hot outside 😩 <<<
CS: indianapolis,in Its so hot outside noone should be working in such amazing weather
CS: tuscaloosa,al @Ri_S_OB It's just so hot outside. https://t.co/uuIU23Bsa8
CS: ruston,la It's so hot outside
CS: skiatook,ok @HunterLuster WHERE ARE YOU?! It's so effing hot outside & I just wanted to swim today.
CS: houston,tx Omg he's really gonna wear all those clothes and a jacket to look good when it's hot outside smh wyd boi
CS: phoenix,az In case you didn't realize, it's really hot outside in Phoenix - https://t.co/yglca4Y5xd
F: tempe . 3044734782
CS: sacramento,ca Dear Southern California: it is n

CS: san antonio,tx It's sooooooo hot outside! Geez
F: united states 757985401
CS: austin,tx Damn it's so hot outside
CS: salt lake city,ut I get that it's hot outside but damn it doesn't have to be freezing in here
F: united states 860180634484039686
CS: cimarron,ks Im pretty jacked up to sit in saftey meetings all day tomorrow. For real, its hot outside.
F: california 1561700640
CS: mcallen,tx From fellow pet owner to pet owner, it's too fucking hot outside to leave your doggo out there
F: ohio  usa 501144463
F: south carolina  usa 2483905243
CS: aurora,il It may be hot outside but keep your freezer cold to avoid food waste! (We assume ice cream isn't listed for obvious… https://t.co/mwXnBJzReN
CS: benton harbor,mi Haven't been to the beach since it's been hot outside 🙄 I really wanna go today , but never have anyone to go with 🤷🏽‍♀️p
CS: fort wayne,in It's hot outside....You need one of these. Tropical Gose- Gose with pineapple, orange, and… https://t.co/AB7eB1miMI
F: remote 18832143

75 hits on page 17 (max_id=871499948814151682)
164 queries remaining.
F: arizona  usa 4711718491
F: the gym 2300390318
CS: ferrysburg,mi You know it's hot outside when you go outside and it's hot
CS: fort worth,tx @TheEliJordan @DustinDietz18 Is anyone going to the games now? It's about to get stupid hot outside, who will go? B… https://t.co/klsYBtOmh0
CS: las vegas,nv it's not even too hot outside, it's just annoyingly dry
F: success 455329922
CS: los angeles,ca Left Jav's flowers in the car. It's so hot outside they're probably gonna be dried up by the time I get back to them 🙃 some friend I am 😂
CS: mercedes,tx Oh my goodness it's so hot outside!!!
KC: twin cities mn It's super hot outside☀️trying to stay cool in this Lacey Dress from @freepeople   🐚💕🐚💕🐚💕🐚💕🐚💕 💕💕… https://t.co/mfY98xcBsI
CS: queens,ny " Stop wearing revealing clo-" STFU! It's hot outside, no women in their right mind wants to wear a space suit in 110 degrees weather!
CS: huntsville,al It's too damn muggy & hot outsid

F: mke/chi 1021304966
F: usa 731707082492252161
CS: boise,id You know it's hot outside when you got outside & it's hot outside
CS: chicago,il Dumb hot outside
F: north carolina  usa 856339807953457152

75 hits on page 19 (max_id=871433015850917889)
161 queries remaining.
CS: new orleans,la It's fucking hot outside
CS: jacksonville beach,fl It's entirely too hot outside
CS: minneapolis,mn @MeatSauce1, it's too hot outside. BTsquaredF2 is on TBS. #powertrip
CS: racine,wi It's so hot outside Ima fuck around go to the broken piers and jump it 😂💯
KC: st louis Hot outside
F: united states 106570187
CS: chicago,il chicago u are WAY too hot outside today
F: bluefield 2369044958
CS: houston,tx It's stupid hot outside ..
KC: chicago Torn between "It's warm outside, I should go do something" and "It's actually really hot outside and my AC feels nice"
CS: springfield,il "It's so hot outside that I've lost the will to live."  - my pregnant wife  #pregnancyproblems
CS: grand rapids,mi So we played i

75 hits on page 1 (max_id=inf)
159 queries remaining.
CS: new york,ny 19 Breezy Outfits For When the Weather Is Just Too Hot to Handle https://t.co/op0MUB3MIB https://t.co/rXjpcvmUzL
CS: las vegas,nv It's hot outside, let me go swimming in your eyes
CS: shawnee,ok When it's hot outside and raining at the same time ..😩 https://t.co/9dUbLTT66v
F: toronto  ontario 736212357584654336
CS: las vegas,nv Watch Hard 4K Porn Videos For Free ➔ https://t.co/9lT4vOIYhU #porn #vip #movies #video #ass #cumtribute https://t.co/Nw5qPopxM7
F: minnesota 860550394006646784
CS: las vegas,nv Watch hard porn videos for free ➔ https://t.co/9lT4vOIYhU #hotgirl #hot #pussy #sext #videos #movies https://t.co/gcyWcApv0x
CS: phoenix,az Hot outside? Grab a refreshing... sewage popsicle? (Yuck; it's not really edible!) https://t.co/uShSeoBhAi https://t.co/3tIPYiKtFP
CS: phoenix,az Hot outside? Grab a refreshing... sewage popsicle? (Yuck; it's not really edible!) https://t.co/vLf057sCty https://t.co/gV4Omkg37f
CS: at

F: new york 16561306
CS: las vegas,nv adult vids in HD get your fREE access ➔ https://t.co/9lT4vOIYhU #nsfw #video #pussy #ass #movies #sexy #swallowcum https://t.co/37vs6zxbK9
F: colorado  usa 1066526090
F:  1145906497
CS: winter park,fl It's hot outside today. I'm slightly bored wanting a cold shower
CS: fontana,ca I feel like it's so hot outside & now I'm debating if I wanna dress like a thot today or put on actual clothes
F: canada 2601553358
F: utah  usa 447291916
F: jamaica 4214813592
F: midwest 362034298
CS: orlando,fl It's too damn hot outside
F: texas  usa 713197429
CS: scottsdale,az It's hot outside 👉 shade your head in style! https://t.co/Ycvp7gkE5x
CS: amarillo,tx It's way hot outside , I'm melting .
CS: austin,tx @nahitscooll It's hot outside!
F: overbrook  philadelphia 825706433278799872
LL: 27.220232,-80.795969 Ok it’s hot outside, so I changed my mind, it can rain if it wants https://t.co/KYNTQMxaMJ
KC: chicago jesus christ it's hot outside 😰
F: florida  usa 87005413715

CS: new york,ny hey! it's the time of the year where it's hot outside and people sweat! wear deodorant and shower! please! thank you!
F: texas 2924716327
KC: chicago il usa https://t.co/DE28LaKIxE
CS: kansas city,mo I know it's hot outside but damn it feel like I like in a damn ice castle lmao I'm about to turn the heat on
CS: las vegas,nv Get Your Vip Access For Free | Best 4K Hard Videos ➔ https://t.co/9lT4vOIYhU #teen #porn #hdporn #boobs https://t.co/Oo7PwbXMir
F: cancun  mx 22742875
F: texas  usa 718843221187493888
CS: kansas city,mo It's hot outside an it looks like she's natural so it's possible that she could have sweated out her leave out 😑😑😑😑
CS: killeen,tx It may be getting hot outside, but don’t expect Killeen’s Pershing Park Pool to open up anytime soon. https://t.co/YUE3MoI7CP
F:  587198022
F: goderich ontario canada 172002568
KC: houston I'm deciding if I should go to pride this year or not it's gonna be so damn hot outside and I got no friends but idc
F: chicagoland usa

CS: cottonport,la It's too damn hot outside.
CS: nashville,tn The weather may be hot outside, but it's cool as Ice Ice Baby at the @nashiconmusic @nashfm1033 @nashfm947... https://t.co/e6xkCMzXeH
GEO: [36.166145, -86.771393] The weather may be hot outside, but it's cool as Ice Ice Baby at the nashiconmusic @nashfm1033… https://t.co/EnKZRemH2z
CS: san antonio,tx i don't care how hard or easy a workout is, if you're doing it when it's hella hot outside it gets like 10 times harder😓☀️🔥
CS: san antonio,tx It's crazy hot outside
F: toronto  ontario 848340199826493441
CS: houston,tx So damn hot outside 😅
CS: san antonio,tx fuckin hot outside
CS: lexington,ky It's hot outside, but the sky is blue. I'd call that a double standard!
CS: minneapolis,mn @Dish_AWD BUT ITS HOT OUTSIDE!!
F: staten island 529078896
F: north carolina 4868361670
CS: wheaton,md It may be hot outside, but you can stay cool at the @WheatonIce  arena! https://t.co/EXBK6v88J6
CS: dallas,tx ⚠️it's getting quite hot outside so

75 hits on page 8 (max_id=872580803871592449)
153 queries remaining.
CS: el centro,ca Everything's a mission when it's this hot outside
CS: west jordan,ut It's fucking hot outside. 98° 😐😥
CS: fresno,ca It's so hot outside cracking open a cold one https://t.co/pTSjCcjPHz
CS: las vegas,nv Get Free Access And Watch Best HD Porn Videos ➔ https://t.co/9lT4vOIYhU #video #boobs #movies #freeporn https://t.co/f0cAAr2P7l
CS: decatur,il Its to hot outside for my own good
F: somewhere in texas 2926413223
F: texas  usa 167878928
F: the backyard 228161955
KC: el paso It's so hot outside, let me float naked in your pool while drinking a 40oz.
CS: houston,tx @Jzacharyyrahcaz You right, us Mexicans were BORN to work lol.. but still to hot outside for me 🤷🏻‍♂️
CS: tampa,fl How to Manage Your Diabetes When It's Hot Outside https://t.co/NQkqNYqW6v https://t.co/UBOdstOQt3
F: toronto 162395712
CS: goldsby,ok It's hot outside, but we've got cold drinks and good music!! Open mic starts at 7 with Dwight Hamli

F: minnesota  usa 146581685
F: nevada 352532564
CS: east peoria,il @JimGaffigan You dropped it. Dropped it like it's hot...outside. Inside, it's nice and cool. That's why the bugs try to get in.
CS: dallas,tx Getting hot outside 3 more hours to go
CS: potomac,md It's getting hot outside. Girls start acting up in this weather https://t.co/zA1BOIhSQk
CS: charlotte,nc @Pruny_thebeagle He's been doing good! Just trying to stay cool while its been hot outside.
CS: austin,tx It's so freaking hot outside!!
F: moon 105005407
CS: pataskala,oh Sheesh I need a gf . It's too hot outside to be single right now
CS: statesville,nc So hot outside.  Can't wait to get home Dx
KC: atlanta georgia usa POLL: #Summer heat got you beat? For #WorkoutWednesday, tell us how you stay fit when it’s hot outside!
F: california  usa 3238031588
CS: new york,ny It's the start of "it's balls hot outside but -10 degrees in your office" season
CS: miramar,fl When it's hot outside, head inside at #AtlanticoAtMiramar! We'v

CS: denton,tx It's hot outside
CS: tucson,az @m_izerd23 Bwhahahaha shut up it's hot outside
GEO: [36.26209904, -115.08537258] To hot outside Dad I'll just stay right here 105' out there..... @ North Las Vegas, North Las… https://t.co/We4z7RnlST
F:  1255989469
CS: rochester,ny real spill I worked out for all them hoes who said I gotta bird chest last summer 😂💯 wait till it get hot outside imma flex without trying
F: washington  usa 856294135795376128
F: south carolina 2612581062
F: nc  usa 27767787
F: oklahoma  usa 745647654261108736
CS: edinburg,tx It's so hot outside!😩😭
CS: refugio,tx It freaking hot outside!!!!
F: haltom city 587844473
CS: round rock,tx It's too hot outside 🔥
CS: dallas,tx listen... it's too damn hot outside
F: maryland 22300781
F: ohio 597987914
CS: phoenix,az Did anyone else know it's hot outside or is that just me?  I might have a magic temp gauge that is giving me... https://t.co/7RpkgiDFFB
CS: wylie,tx Really don't wanna walk in this store.. too damn hot outside

CS: los altos,ca It's getting hot outside and every year when Summer rolls in the murder rate skyrockets in St. Louis. Chicago is a…… https://t.co/AnCE1ap850
CS: sierra vista,az Shout out to the Arizona sun for defeating me, I rarely wear makeup now since it's WAY TO HOT OUTSIDE
F: jersey 1905150278
F:  105390401
CS: college station,tx @LeifTheTank @Daegyn @CAtoDC I don't think you know what "it's too hot outside" truly means anyway.
F: charlottesville 22919204
CS: las vegas,nv Get Your Free Access To Best Porn Vids In 4K ➔ https://t.co/9lT4vOIYhU #hugetits #hot #freeporn #vip https://t.co/suakeJkmxH
F: drew 714679395584253954
F: playing volleyball 599932448
CS: houston,tx It's always like that when it's hot outside.. https://t.co/9GB3YIRpCI
CS: sacramento,ca It's to hot outside for a playa.. I need to be on a beach somewhere 🏝
CS: omaha,ne @MeanStreetsOMA Well, it is hella hot outside....
F: tx 1445318936
CS: las vegas,nv Me: Wowie it's so hot outside. Also me: https://t.co/bUSMtg5FYc

KC: las vegas When it's this hot outside cool off with a cold Drink! https://t.co/NELrgo7yLv
F: winnipeg  manitoba 3329543117
CS: mesa,az It's Monday and it's hot outside. Come let our drink specials & misted patio keep you cool! 1/2 price liquor & $2.5… https://t.co/IXMOZi1mr8
CS: pahrump,nv Whoops, it's that hot outside 😅 https://t.co/Se0Mz0hjqM
CS: compton,ca Go fetch me some water @Vince_Steel, it's hot outside
CS: lawrence,in My God!! It's So Hot Outside, Well Not Really? My House Is Just Cold Lol
CS: tucson,az Its super hot outside, make sure to wear a hat, we have a huge selection.,
F: tx 3241972296
CS: palm desert,ca It's hot outside - spash into PDAC! #pdswimlife https://t.co/ohAtEiwbEY
CS: baton rouge,la It's hot outside bruh or it's just me
F: tx 613706459
CS: clay,ny @kerrycalderon awww.  hot outside?  not here.  lol.  it's been raining here
F: california 2423655005
F: arizona  usa 4884245606
CS: san antonio,tx It's too hot outside to be alive rn
CS: baton rouge,la FOR THE 

F: iowa  usa 3146761261
CS: west valley city,ut Guys! Rosa has been missing since 2 pm. We can not find her anywhere. She is old and it's hot outside. Please if yo… https://t.co/xwaGg3Rqos
F: uptown  new orleans 3034062821
F: regina  saskatchewan 352342907
F: minnesota  usa 822947066229846017
F: east la 882871710

75 hits on page 17 (max_id=871513922385227775)
144 queries remaining.
CS: phoenix,az ITS SOOOOO FUCKING HOT OUTSIDE
CS: westboro,wi Fitting the drawer and runners on this lacewood and maple jewelry box. It's hot outside but my… https://t.co/WukZQiHpKr
CS: chicago,il It's too mf hot outside 😩
CS: indianapolis,in Davyn: "dont walk its SO HOT OUTSIDE" *me gets home*.... its 86 degrees in this bitch 😐 🤦🏽‍♀️ why wouldnt this nigga turn the air on
CS: avondale,az @_thesaucebosss I honestly have no idea 😂 it was so hot outside I was totally out of it lol
CS: chicago,il It's really hot outside. How many showers should I take today?
CS: chicago,il It's disgustingly hot outside
F: jama

F: the castle 325203607

75 hits on page 19 (max_id=871438159510478850)
141 queries remaining.
CS: austin,tx Fuck it.... it's too hot outside... this how I'm comin today https://t.co/4fwkIlXS34
F: kentucky  usa 768680936930156545
F: fl 66740131
CS: madison,wi It's hot outside, the air conditioner is on and working at The Bos Meadery.  Open mic today 2-5.  First 5... https://t.co/UIBWRFiap9
F: virginia  usa 844395289041666048
CS: houston,tx @iniguezmari_ Good mama because it's hot outside and he needs air condition 😤
F: new york  usa 2899044732
CS: los angeles,ca It's hot outside 🙄 everybody be safe
CS: madison,wi It's hot outside, the air conditioner is on and working at The Bos Meadery.  Open mic today 2-5.  First 5... https://t.co/LePvmZrHBb
F: ohio 239062772
F: if only i was cool & in gc's 4687965025
F: mke/chi 1021304966
F: usa 731707082492252161
CS: boise,id You know it's hot outside when you got outside & it's hot outside
CS: chicago,il Dumb hot outside
F: north carolina  usa 856

75 hits on page 1 (max_id=inf)
139 queries remaining.
CS: las vegas,nv It's hot outside, let me go swimming in your eyes
CS: shawnee,ok When it's hot outside and raining at the same time ..😩 https://t.co/9dUbLTT66v
CS: las vegas,nv Watch Hard 4K Porn Videos For Free ➔ https://t.co/9lT4vOIYhU #porn #vip #movies #video #ass #cumtribute https://t.co/Nw5qPopxM7
CS: las vegas,nv Watch hard porn videos for free ➔ https://t.co/9lT4vOIYhU #hotgirl #hot #pussy #sext #videos #movies https://t.co/gcyWcApv0x
CS: phoenix,az Hot outside? Grab a refreshing... sewage popsicle? (Yuck; it's not really edible!) https://t.co/uShSeoBhAi https://t.co/3tIPYiKtFP
CS: phoenix,az Hot outside? Grab a refreshing... sewage popsicle? (Yuck; it's not really edible!) https://t.co/vLf057sCty https://t.co/gV4Omkg37f
F: iowa  usa 827738064264925185
CS: lancaster,ca Went into the mall and it was hot outside walk out of the mall and it's freezing outside 🙃🙃
CS: maricopa,az So HOT outside Mello had to put on a Fire hat. Cl

F: out the way... 1733776549
CS: sacramento,ca It's not even hot outside and my neighbors have their AC going 24/7, are they polar bears?
F: united states 2763599017
CS: san antonio,tx #Fitness idea: When it is too hot outside, stay indoors and check the Internet for light exercise ideas.... https://t.co/yUihznGFFz
CS: dallas,tx It's so hot outside I don't even wanna eat.
CS: las vegas,nv Watch Hard Porn HD Videos For Free ➔ https://t.co/9lT4vP0z9s #fucking #videos #hot #boobs https://t.co/i508uotpSu
F: california  usa 2366351558
F: home 700667760171298818
CS: houston,tx Y'all securing the bag but cant secure ya leave out bitch it's hot outside & you got a hoodie on ya head
CS: kansas city,mo I know it's hot outside but damn it feel like I like in a damn ice castle lmao I'm about to turn the heat on
CS: las vegas,nv Get Your Vip Access For Free | Best 4K Hard Videos ➔ https://t.co/9lT4vOIYhU #teen #porn #hdporn #boobs https://t.co/Oo7PwbXMir
CS: kansas city,mo It's hot outside an it lo

F: edmonton  alberta 1638376668
CS: houston,tx Clap back season , it's starting to get hot outside, leave people alone . https://t.co/dmRD4wCfY8
F:  1244886295
CS: santa barbara,ca If you're hot outside.. and need water.. what makes you think... that your dog .. doesn't need water... FRESH water...
CS: las vegas,nv watch thousands of videos featuring 4K extreme porn ➔ https://t.co/9lT4vOIYhU #nsfw #pornstar #movies #amateur https://t.co/rdDQWClig4
CS: hood river,or Baby it's HOT outside.... https://t.co/4IClmXlkuR
CS: muscatine,ia Looks like next week will be very hot outside. Don't wait until next week to get hydrate. Hydrate now!
CS: las vegas,nv Summer classes are great for when you want to get coughed in constantly by other students, but it's hot outside
F: kamloops  bc  canada 15478970
CS: provo,ut Hey @BYU I don't think it's safe to hold class when it's this hot outside
CS: las cruces,nm I really ❤️ summer evenings when its still hot outside I smell someone smoking
KC: houston it

75 hits on page 6 (max_id=872508116478328833)
136 queries remaining.
CS: austin,tx It's pretty (hot) outside.
CS: arlington,tx I hate getting ready when it's hot outside because I sweat so much by the time I'm done
F: venus 2883983289
CS: houston,tx It's too fuckin hot outside 😡😒
CS: center,tx it's too hot outside
F:  70841255
F: nevada 352532564
CS: dallas,tx Getting hot outside 3 more hours to go
CS: austin,tx It's so freaking hot outside!!
F:  3060139561
CS: conway,ar It's too hot outside for alllatt.
F: utopia 2322918493
CS: las cruces,nm Its only 9am and I'm still in bed, but its already so fu king hot outside
CS: austin,tx It's hot outside, drink water
F: cc  tx. 734287297
CS: minneapolis,mn It's already so hot outside and it's only 10am omg
KC: san antonio it's so hot outside
CS: scottsdale,az It's HOT outside and our prices are melting...Nove' Semi Annual Sale starts today save up to 70% off on your... https://t.co/J6Xs6MLLuq
F: louisiana  usa 3223319983
F: usa 100986964
F: lag

KC: la @TrelaTTN Enjoying your summer! It's been so hot outside so video games are even more fun cuz it's not sweaty work!… https://t.co/roT5OYZ7KJ
F: stl 4015514173
F: san marvelous  texas 2596787382
KC: dallas tx  ks It's hot outside 🙄
CS: houston,tx Back to rocking my puffs. It's too hot outside.
F: méxico 466868343
CS: kent,wa IT IS SO HOT OUTSIDE I'M DYING.
CS: new orleans,la It's really fucking hot outside can't even enjoy these blunts 😒
CS: houston,tx It's so hot outside like I'm burning up!
CS: bellevue,wa "It's nice outside!"  No, it's sunny and hot outside. Those are two very different things, folks.
CS: austin,tx Cuz it's hot outside and I'm grinding of course
CS: houston,tx No reason it should be that hot outside
F:  140199963
CS: portland,tx It is so hot outside
KC: st louis mo @ALL_SmiLeS_xOXo cause its hot outside
CS: horn lake,ms It's so hot outside.. wheeeww
CS: broken arrow,ok It's so hot outside
KC: garland I really feel like going outside and run, but it's sooo hot 

KC: kansas city So hot outside. When will it cool back down a bit? https://t.co/M2NtEloeYJ
CS: kansas city,mo It's so hot outside.. me & pho will be at the pool today 🎀
CS: houston,tx My windows need to be tinted it's too damn hot outside
F: south texas 820435111
F: united states 863262061341663232
KC: omaha @CitizenRyan82 i know it’s not that hot outside, but man, that radiant heat off the parking lot and the car sitting… https://t.co/13TPzbofHL
CS: brownsville,tx It is way too hot outside
CS: riverside,ca Can someone tell me if it's hot outside I'm too lazy to get up
CS: alexandria,mn Couch to 5k: Baby, It's HOT Outside -  https://t.co/ixyWmVkOHt
KC: las vegas It's so hot outside!! 🌞
F: united states 4723499898
CS: la feria,tx It's so hot outside😩
F: texas  usa 1717128186
CS: corpus christi,tx Bright day out side, stay cool little birds. It's hot outside 💋 https://t.co/ymmF4XfU7N
KC: las vegas When it's this hot outside cool off with a cold Drink! https://t.co/NELrgo7yLv
F: winnipeg 

CS: las vegas,nv It's tooooo fucking hot outside !
CS: belton,tx It is very very hot outside
F:  408947330
F: united states 3096715238
CS: salt lake city,ut It's way too hot outside so I'm gonna take a nap.
CS: edinburg,tx holy smokes, it's hot outside
CS: orion,il Someday when i have my own house and family, i'm cranking the A/C the SECOND it get's hot outside. My kids are never sweating in the house
CS: denver,co Waiting on @dazhaloveeee  like come on B, it's hot outside 😅
F: sfv/la 27831738
F: united states 861038361942863873
CS: sinton,tx It's hot outside, I hate it. :(
F: wit yo bitch 736415577816731648
F: o'connor 719723483287187457
F: united states 871206583836999682
CS: corpus christi,tx ITS SO HOT OUTSIDE. It's perfect for swimming. 🙆‍♂️
CS: reno,nv Good thing it's hot outside. I spilled a gallon of water on the front of my pants right before I started lunch. #fml
F: united states 2582911394
F: the castle 325203607
CS: austin,tx Fuck it.... it's too hot outside... this how I'm

F: united states 773766956923125762
CS: memphis,tn Baby Its Too Damn Hot Outside ..
CS: houston,tx It may be getting hot outside, but you can still have cool styles for less! Shop our 50% off Clearance sale NOW! https://t.co/kxgXjYLujq
CS: san antonio,tx Damn it's hot outside. I'm sweating like a whore in church in the shade...
F: cuba 1118360270
CS: houston,tx it is so hot outside today wtf
CS: iowa city,ia Would be a good day... to do nothin!  Instead I'm trying to do everything😩😩 and it's blazing hot outside
F: st. louis mo 26958920
F: jalisco 703588026
GEO: [49.97607127, -125.23596953] Hot outside #nostresshere @ Willow Point Sportsplex https://t.co/kjWtn1jCef
CS: san antonio,tx It is TOO hot outside I don't think I'm ready for summer, help
CS: houston,tx Like damn man my whole attitude is not going and it's fucking  hot outside
KC: st louis mo It's too damn hot outside
CS: fairfield,ia It's hot outside, come join us on the patio for some cold drinks, we have mojitos , adult lemona

CS: houston,tx i need to get a short wig asap. it's toooo hot outside to be having hair on my neck 💀💀💀
F: stl 612767707
F: kansas  usa 800062349549965313
KC: memphis tn    courtney  Why is it so fucking hot outside.
F: north dakota 2928950863
KC: st louis mo it's so hot outside 😩
F: winnipeg  canada 15102532
F: redondo beach 26761011
KC: houston its so sticky & hot outside.
F: probably at ihop 1315494600
CS: oklahoma city,ok When it's hot outside, our favorite treats are wine ice cream from @wew_okc or iced coffee from @coffeeslingers
CS: farmington,mn It's too hot outside to wear jeans o3o;;;
CS: college station,tx Just left the track. Damn it's hot outside but it felt sooo good!
KC: st louis mo It's "dont say anything to me" type of hot outside today.
F: audrey 4785637592
CS: mesa,az It was so hot outside when I went on the patio to eat my lunch I got disgusted and walked back in
F: vancouver  b.c. 16152163
CS: omaha,ne I know it's June and it's starting to get hot outside, but it's 

29 hits on page 17 (max_id=870346407261200383)
124 queries remaining.
F: dallas/denton 105177817
CS: pflugerville,tx You know it's hot outside when you walk outside and it's hot .
F: united states 826510287868162049
CS: oklahoma city,ok Bitch it's hot outside
F: united kingdom 3295671478
F: somewhere in a gym 2944112929
F: the gym 42734572
CS: new orleans,la So fucking hot outside.
CS: lawton,ok Even though it may not seem that hot outside yet, never leave your child or animal in a car! #safety #summer https://t.co/VmiZWEi9NA
CS: murray,ut It's so fucking hot outside
CS: salt lake city,ut Summer colds are the worst. I can't tell if I have a fever or if it's just hot outside.
F: oklahoma 45497343
CS: phoenix,az Me: I love summer mornings Neighbor: It's actually not summer yet Me: *long stare* school is out and it's hot outside! It's effing summer!
CS: hopkins,mn @TheCrystai it's so hot outside ☀️😎✌🏻gorgeous day Tristan should I order another fidget spinners online yes or no saw you got 

75 hits on page 2 (max_id=873268654716137471)
121 queries remaining.
CS: san francisco,ca Almost too hot outside for a royal blue blazer, @KarlTheFog. Almost.
F: bmt  tx 2824290643
F: michigan  usa 286163003
CS: pearsall,tx It's so hot outside 😩
F: vancouver  british columbia 787132733076090880
CS: mcallen,tx It's freakin hot outside  #damn
F: somewhere in the mountains 17013270
CS: omaha,ne Chauncey gone kill me for my outfit. But its hot outside soo it might be worth the death 😂😂🤷🏾‍♀️
KC: san antonio Is too damn hot outside, smoking inside I guess 🙄🙃
CS: paradise,ca Its so hot outside omg  #NicoleforUGirlJune13th
F: dregangchas✨ 488926183
F: louisiana  usa 810663891550302209
F: united states 432088222
CS: san bernardino,ca [F4M] Baby, It’s Hot Outside! [PREVIEW] [Erotic] [GFE]: https://t.co/gmZR8ZFlGb via @YouTube
KC: kansas city It's hot outside, folks. - said the car salesmen that just spent 40 minutes trying to chase down a car on our massive back-lot.
CS: denver,co It's so damn h

F:  3038782688
CS: san benito,tx It's so hot outside...and I'm supposed to cut the grass! No way
CS: las vegas,nv It's so hot outside😩😩
CS: las vegas,nv Watch Hard Porn Videos For Free ➔ https://t.co/9lT4vOIYhU #amateur #videos #tits #porn https://t.co/zw4P4q51D2
CS: los angeles,ca It's so hot outside bra😩
CS: san antonio,tx It's so hot outside!
CS: baytown,tx It is so hot outside omg
CS: las vegas,nv @mikestokoe its sooooo hot outside!!
CS: allen,tx Schools out and it's hot outside, sound like it's time for some cookies & ice cream at Grif's! #cookies #icecream… https://t.co/iptB14E0kh
CS: dallas,tx It's hot outside 😍
CS: san antonio,tx @wadduperikaa Its too hot outside for you to deal with that. I can help you get into something that you don't have… https://t.co/7o3u9i2Z2V
F: missouri  usa 850044547400957952
F: iowa  usa 2941646271
F: usa 393594798
F: area 51 293846156
CS: new orleans,la It's so hot outside
GEO: [45.11025, -93.35236] "Do you want to build a snowman" ~ frozen~  It may

CS: phoenix,az Join us @StateFarm Volunteer Fair in Tempe AZ. Plus it's hot outside and we have air conditioning. #HereToHelp… https://t.co/7W6vryTZrJ
CS: kyle,tx too damn hot outside
CS: kansas city,mo When it's hot outside, setting the AC to 65 degrees for perfect sleeping. #SNEmployee https://t.co/6ywUCOkAyt https://t.co/rQtAWR6uyv
F: bontemps  la 388516692
CS: austin,tx Personally, I really enjoy the drive from Austin to NOLA <3 It'll be even nicer in October when it's not scorching hot outside.
F: tx 1115478714
CS: houston,tx It is dead hot outside... Got me riding with no shirt today.
CS: eugene,or Lectures halls blast the AC when it's hot outside, so it's too cold inside but too hot outside. There's no in between 😅
CS: houston,tx It's too hot outside
CS: dallas,tx Post #run lunch is a green smoothie and pretzel chips bc it's so hot outside I can't manage cooking.
CS: whitehouse,tx @jake_p84 This dude just ordered coffee. Bro, it's so fucking hot outside. That would make me sweat

F: stl 4015514173
F: san marvelous  texas 2596787382
KC: dallas tx  ks It's hot outside 🙄
CS: houston,tx Back to rocking my puffs. It's too hot outside.
F: méxico 466868343
CS: kent,wa IT IS SO HOT OUTSIDE I'M DYING.
CS: new orleans,la It's really fucking hot outside can't even enjoy these blunts 😒
CS: houston,tx It's so hot outside like I'm burning up!
CS: bellevue,wa "It's nice outside!"  No, it's sunny and hot outside. Those are two very different things, folks.
CS: austin,tx Cuz it's hot outside and I'm grinding of course
CS: houston,tx No reason it should be that hot outside
F:  140199963
CS: portland,tx It is so hot outside
KC: st louis mo @ALL_SmiLeS_xOXo cause its hot outside
CS: horn lake,ms It's so hot outside.. wheeeww
CS: broken arrow,ok It's so hot outside
KC: garland I really feel like going outside and run, but it's sooo hot outside ! 😔
F: dallas/denton 105177817
CS: dallas,tx It's too hot outside...
F: carman mb 436299973
CS: las vegas,nv Even when it's hot outside, we 

CS: houston,tx My windows need to be tinted it's too damn hot outside
F: south texas 820435111
F: united states 863262061341663232
KC: omaha @CitizenRyan82 i know it’s not that hot outside, but man, that radiant heat off the parking lot and the car sitting… https://t.co/13TPzbofHL
CS: brownsville,tx It is way too hot outside
CS: riverside,ca Can someone tell me if it's hot outside I'm too lazy to get up
CS: alexandria,mn Couch to 5k: Baby, It's HOT Outside -  https://t.co/ixyWmVkOHt
KC: las vegas It's so hot outside!! 🌞
F: united states 4723499898
CS: la feria,tx It's so hot outside😩
F: texas  usa 1717128186
CS: corpus christi,tx Bright day out side, stay cool little birds. It's hot outside 💋 https://t.co/ymmF4XfU7N
KC: las vegas When it's this hot outside cool off with a cold Drink! https://t.co/NELrgo7yLv
CS: mesa,az It's Monday and it's hot outside. Come let our drink specials & misted patio keep you cool! 1/2 price liquor & $2.5… https://t.co/IXMOZi1mr8
CS: pahrump,nv Whoops, it's 

F: wit yo bitch 736415577816731648
F: o'connor 719723483287187457
F: united states 871206583836999682
CS: corpus christi,tx ITS SO HOT OUTSIDE. It's perfect for swimming. 🙆‍♂️
CS: reno,nv Good thing it's hot outside. I spilled a gallon of water on the front of my pants right before I started lunch. #fml
F: united states 2582911394
F: the castle 325203607
CS: austin,tx Fuck it.... it's too hot outside... this how I'm comin today https://t.co/4fwkIlXS34
CS: houston,tx @iniguezmari_ Good mama because it's hot outside and he needs air condition 😤
F: if only i was cool & in gc's 4687965025
F: usa 731707082492252161
CS: boise,id You know it's hot outside when you got outside & it's hot outside
CS: new orleans,la It's fucking hot outside
CS: minneapolis,mn @MeatSauce1, it's too hot outside. BTsquaredF2 is on TBS. #powertrip
KC: st louis Hot outside
F: united states 106570187
CS: houston,tx It's stupid hot outside ..
F: canada 375442181
CS: liberal,ks @Gods___gift @GanjaGoddessbae @caamaachhoo

F: jalisco 703588026
GEO: [49.97607127, -125.23596953] Hot outside #nostresshere @ Willow Point Sportsplex https://t.co/kjWtn1jCef
CS: san antonio,tx It is TOO hot outside I don't think I'm ready for summer, help
CS: houston,tx Like damn man my whole attitude is not going and it's fucking  hot outside
KC: st louis mo It's too damn hot outside
CS: fairfield,ia It's hot outside, come join us on the patio for some cold drinks, we have mojitos , adult lemonade and Moscow mules. https://t.co/Uz1KT7Gr3j
CS: bloomington,ca It's too fuckin hot outside!
CS: san antonio,tx Too Hot Outside? Book a Penguin Interaction today to cool off and meet some new friends! Pass members save 20%! https://t.co/QJAC11VGO8
CS: tucson,az It's 9:30 & it's already hot outside 😩 #AZProbz
F: astoria 30923399
KC: st louis mo Playing that hot outside, freezing cold inside game--yay for June! Today I layered a basic… https://t.co/SBhQli6WtA
CS: houston,tx It's so fucking hot outside ITS RIDICULOUS
CS: las vegas,nv it's 

F: vancouver  b.c. 16152163
CS: omaha,ne I know it's June and it's starting to get hot outside, but it's not summer until I hear the ice cream truck driving down the street.
CS: visalia,ca It's too hot outside 😭
CS: hopkins,mn @TheCrystai it's so hot outside in the upper 80's☀️😎staying inside where it cool at gym workout u are getting me motivated 2💪🏼🥊workout again
F: us of a 568777982
F: united states 717773405425115136
F: alberta  canada 1650761390
CS: waco,tx Soon, it'll be super hot outside and providing fresh, clean water is very important! We proudly sell JUG... https://t.co/UvM6b9pGXy
CS: mission viejo,ca why is it so mf hot outside 🙁
KC: st louis When it's too hot outside to go for a walk but you decided to wear your fur coat anyway https://t.co/9Ozw3qL0EF
CS: avondale,az Bruh its hot outside already
F: humboldt  sk 337282307
CS: portland,or @evanempdx "People are saying, 'it's hot outside', and I'm just like, 'that's the sun'... idk what everyone's in a fuss about."
CS: des mo

9 hits on page 17 (max_id=870217121212641279)
106 queries remaining.
F: devon 423625037
GEO: [34.697781, -118.143675] ☀️Welcome June & Summer!☀️ It's gonna get HOT outside!🔥 Keep your skin protected, get your… https://t.co/uCz1aPk3Oi
CS: las vegas,nv Watch Hard Porn Videos For Free ➔ https://t.co/9lT4vP0z9s #amateur #videos #tits #porn https://t.co/zw4P4qmCuA
CS: dallas,tx its HOT outside, time to shave head
CS: lafayette,la Why tf is it so hot outside
F: tennessee  usa 704315318118830080
F: united states 823476844452118529
F: canada 54085062
KC: dallas it's too fucking hot outside   i can't do it

0 hits on page 18 (max_id=869976105851080703)
106 queries remaining.
Out of tweets at this location.
Balancing multi-location search results.
maximin: 2.28661410997e+15
16 cutoffs for loc 0
0 cutoffs for loc 1
350 cutoffs for loc 2
336 cutoffs for loc 3


### Now go to R to finish processing

In [7]:
scheduled

False

In [29]:
print outfile

good+morning.csv
